source/agora/utils/HumanReadable.d

```diff
@@ -0,0 +1,284 @@
+/*******************************************************************************
+
+    Defines method to format types to their human readable representation
+
+    This approach allows to take the string formatting out of the type,
+    simplifying the implementation and reducing dependencies.
+    It is assumed that a human-readable representation will not rely on
+    non-visible data, or data that cannot be accessed without mutation.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.utils.HumanReadable;
+
+import agora.common.Amount;
+import agora.common.Data;
+import agora.common.crypto.Key;
+import agora.consensus.data.Block;
+import agora.consensus.data.Transaction;
+
+import std.algorithm;
+import std.format;
+import std.range;
+
+/// Ditto
+public auto humanReadable (T) (const ref T input)
```

Comment:
In many libraries this is called "pretty" printing. So something like `toPretty` is short and concise. But you can keep `humanReadable` too if you want. I'm just thinking of making this function shorter to write.